nntrainer/compiler/loss_realizer.cpp

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2022 seongwoo <mhs4670go@naver.com>
+ *
+ * @file loss_realizer.h
+ * @date 4 May 2022
+ * @brief NNTrainer graph realizer which remove loss layer for inference
+ * @see	https://github.com/nnstreamer/nntrainer
+ * @author seongwoo <mhs4670go@naver.com>
+ * @bug No known bugs except for NYI items
+ */
+#include <algorithm>
+#include <cassert>
+#include <connection.h>
+#include <layer_node.h>
+#include <loss_realizer.h>
+#include <nntrainer_error.h>
+#include <nntrainer_log.h>
+#include <set>
+#include <stdexcept>
+#include <string>
+#include <unordered_map>
+
+namespace nntrainer {
+
+static constexpr size_t SINGLE_IN_IDX = 0;
+
+GraphRepresentation
+LossRealizer::realize(const GraphRepresentation &reference) {
+  /// @todo support more loss layers
+  /// @note Some layers need to consider not removing all semantics.
+  /// For example, When CrossEntropySigmoidLossLayer needs to be removed,
+  /// sigmoid computation shouldn't be removed.
+  static const std::set<std::string> loss_type = {"mse"};
+  std::unordered_map<std::string, LayerNode *> existing_nodes;
+  std::vector<LayerNode *> loss_layers;
+
+  std::transform(
+    reference.begin(), reference.end(),
+    std::inserter(existing_nodes, existing_nodes.end()),
+    [](auto &node) { return std::pair(node->getName(), node.get()); });
+
+  for (auto &node : reference) {
+    if (loss_type.find(node->getType()) != loss_type.end()) {
```

Comment:
Consider requireLabel function of layer.
According to the document of requireLabel, it means, for now, that it is endpoint of a graph(numOutlayers == 0).
So all of loss layer will return true for requireLabel

Reply:
@lhs8928 Thank you for your information. The thing that you said is exactly what I wanted. But it will be one of assert condition because we should list supported loss layer rather than supporting all of them. Please refer to above @note section.
```c++
/// @note Some layers need to consider not removing all semantics.
/// For example, When CrossEntropySigmoidLossLayer needs to be removed,
/// sigmoid computation shouldn't be removed.
```